sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -221,14 +242,11 @@ StreamAppendClient getStreamAppendClient(boolean lookupCache) {
               if (lookupCache) {
                 this.streamAppendClient =
                     APPEND_CLIENTS.get(
-                        streamName,
-                        () ->
-                            datasetService.getStreamAppendClient(
-                                streamName, descriptorWrapper.descriptor));
+                        getStreamAppendClientCacheEntryName(), () -> createStreamAppendClient());
```

Comment:
getStreamAppendClientCacheEntryName doesn't necessarily return the stream name we used

Reply:
It will also be broken if you have two sinks in the pipeline, one using the default stream and one not.

Reply:
I can see the problem of mixing the default with dynamically gen streams, I will change that.